### PR TITLE
Change request login to use user id

### PR DESCRIPTION
**Pull request by TimDiekmann**
r? @wencke-lm

### CODE REVIEW

File: app/views/login/__init__.py [hunk 1]
```diff
@@ -26,26 +26,14 @@ def load_user(id):
 @login_manager.request_loader
 def load_user_from_request(request):
     token_id = request.headers.get('Authorization') or request.args.get('token')
-    if not token_id:
+    user_id = request.args.get('user') or request.headers.get('user')
+    if token_id is None or user_id is None:
+        current_app.logger.warning(f"Loading user from request failed. user: `{user_id}`, token: `{token_id}`")
```
Comment (TimDiekmann) [inline app/views/login/__init__.py:31]:
```suggestion
```

File: app/views/login/__init__.py [hunk 2]
```diff
@@ -26,26 +26,14 @@ def load_user(id):
...
         return None
 
-    current_app.logger.debug(f"loading user from token {token_id}")
+    current_app.logger.info(f"loading user `{user_id}` from token `{token_id}`")
```
Comment (TimDiekmann) [inline app/views/login/__init__.py:34]:
```suggestion
    current_app.logger.debug(f"loading user `{user_id}` from token `{token_id}`")
```

File: app/views/api/users.py [hunk 3]
```diff
@@ -45,8 +47,24 @@ def get(self, args):
     @blp.response(201, UserSchema.Response)
     @blp.login_required
     def post(self, item):
-        """Add a new user"""
-        return UserSchema().post(item)
+        """Add a new user
+
+        The token is required to has registrations left and a room associated"""
```
Comment (janagoetze) [inline app/views/api/users.py:52]: to have?

File: app/views/api/users.py [hunk 4]
```diff
@@ -45,8 +47,24 @@ def get(self, args):
...
+
+        db = current_app.session
+        token = db.query(Token).get(item['token_id'])
+
+        if token.registrations_left == 0:
+            abort(UnprocessableEntity, errors=dict(token_id='No registrations left for given token'))
```
Comment (wencke-lm) [inline app/views/api/users.py:58]: Do not forget, to adapt the 'errors' keyword here. Our custom abort does no longer have an 'errors' keyword.

**Pull request merged**